### PR TITLE
Fix cosmosdb pagination bug

### LINKED ISSUE #6346: Bug: CosmosDB iterator may crash after fetching more items
Consider the following keys under some partition: `1,2,3,4`. CosmosDB client scans with page size 3.
First response returns items `1,2,3` and indicates there are more items with `HasMore` property.
If item `4` was deleted, the next request using the continuation token will fetch 0 items.
CosmosDB store doesn't handle that case properly and crashes on nil pointer.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #6346 6346

### CODE REVIEW

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Nothing about the code:
1. How does the bug manifests?
2. Can we link a an issue to this PR?
3. Can we add a unit test to catch this scenario?

**Comment by itaiad200** [comment]
> Nothing about the code:
> 
> 1. How does the bug manifests?
> 2. Can we link a an issue to this PR?
> 3. Can we add a unit test to catch this scenario?

Thanks, fixed all of the above. Added a test and tested locally against a live CosmosDB account.

File: pkg/kv/kvtest/store.go [hunk 1]
```diff
@@ -420,6 +420,46 @@ func testStoreScan(t *testing.T, ms MakeStore) {
...
+				t.Fatal("Value is nil while scan item", len(entries))
+			}
+			if !bytes.HasPrefix(ent.Key, samplePrefix) {
+				break
+			}
+			entries = append(entries, *ent)
+		}
+		if err := scan.Err(); err != nil {
+			t.Fatal("scan ended with an error", err)
+		}
```
Comment (N-o-Z) [inline pkg/kv/kvtest/store.go:454]:
```suggestion
		require.NoError(t, scan.Err(), "scan ended with an error")
```

**Review by N-o-Z — APPROVED** [review]
Thanks, approved with 2 suggestions (always prefer one liners 😄 )

**Pull request merged**